Modules/FindNetCDF.cmake

```diff
@@ -231,6 +243,10 @@ foreach( _comp IN LISTS _search_components )
       if( NOT _comp MATCHES "^(C)$" )
         target_link_libraries(NetCDF::NetCDF_${_comp} INTERFACE NetCDF::NetCDF_C)
       endif()
+      if(NetCDF_PARALLEL)
+	find_package(MPI REQUIRED COMPONENTS ${_comp})
+	target_link_libraries(NetCDF::NetCDF_${_comp} INTERFACE  MPI::MPI_${_comp})
```

Comment:
I don't think this is appropriate in `FindNetCDF.cmake`.
I think the application using NetCDF should be responsible for linking with the MPI library.
e.g.
in wgrib2 or any other application e.g. chgres_cube, one could do one of two things:
1.  in the application `CMakeLists.txt`
```
if(NetCDF_PARALLEL)
  find_package(MPI REQUIRED COMPONENTS C)
  target_link_libraries(app/lib INTERFACE|PUBLIC|PRIVATE MPI::MPI_C)
endif()
```
I think only the C library is required, but I will have to test that.

2. explicitly set CC and FC with the MPI wrapper e.g.
```
export CC=mpiicc
export FC=mpiifort
```
etc. or whatever the compiler is instead of e.g. `CC=icc, FC=ifort`

Reply:
I'm not really against either of those. In general, `find_package` links in its dependencies, but I could see it being different for MPI.

Reply:
If a serial (non-MPI) application depends on (parallel) netcdf library, it will be much simpler to not need to add `if(NetCDF_PARALLEL) .... endif()` logic in application's CMakeLists.txt or require `CC/FC` to point to mpi wrapper scripts.  Such program should not need to know anything about internal implementation/dependencies of netcdf.

Reply:
I agree with @DusanJovic-NOAA and @kgerheiser that if netCDF depends on MPI, it needs to automatically configure the MPI dependencies for the application that it is used in.